airbyte-integrations/connectors/source-jira/source_jira/client.py

```diff
@@ -92,3 +90,170 @@ def health_check(self) -> Tuple[bool, str]:
+
+    def _get_workflow_scheme_keys(self):
+        if not self._workflow_scheme_keys:
+            for workflow in self.lists(name="workflow_schemes", **ENTITIES_MAP.get("workflow_schemes")):
+                self._workflow_scheme_keys.append(
+                    {
+                        "id": workflow.get("id"),
+                    }
+                )
+        return self._workflow_scheme_keys
+
+    def _get_custom_fields(self):
+        for field in self.lists(name="issue_fields", **ENTITIES_MAP.get("issue_fields")):
+            if field.get("custom"):
+                yield field
+
+    def _get_projects_related(self, name, url_name="", query_param="key"):
+        url_name = url_name or name
+        project_keys = self._get_project_keys()
+        for project_key in project_keys:
+            query_value = project_key.get(query_param)
+            for item in self.lists(name=f"project_{name}", url=f"/project/{query_value}/{url_name}", **ENTITIES_MAP.get(f"project_{name}")):
+                yield item
+
+    def stream__avatars(self, fields):
+        avatar_types = ("issuetype", "project", "user")
+        avatar_configs = ENTITIES_MAP.get("avatars")
+        for avatar_type in avatar_types:
+            for data in self.lists(name="avatars", url=f"/avatar/{avatar_type}/system", **avatar_configs):
+                yield data
+
+    def stream__filter_sharing(self, fields):
+        filter_sharing_configs = ENTITIES_MAP.get("filter_sharing")
+        for filter_item in self.lists(name="filters", **ENTITIES_MAP.get("filters")):
+            filter_item_id = filter_item.get("id")
+            for permission in self.lists(name="filter_sharing", url=f"/filter/{filter_item_id}/permission", **filter_sharing_configs):
+                yield permission
+
+    def stream__issue_comments(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
+            for comment in self.lists(name="issue_comments", url=f"/issue/{issue_key}/comment", **ENTITIES_MAP.get("issue_comments")):
+                yield comment
+
+    def stream__issue_custom_field_contexts(self, fields):
+        for field in self._get_custom_fields():
+            for context in self.lists(
+                name="issue_custom_field_contexts",
+                url=f"/field/{field.get('id')}/context",
+                **ENTITIES_MAP.get("issue_custom_field_contexts"),
+            ):
+                yield context
+
+    def stream__issue_properties(self, fields):
+        issue_keys = self._get_issue_keys()
+        for item in issue_keys:
+            issue_key = item.get("key")
+            for issue_property_key_item in self.lists(
+                name="issue_property_keys", url=f"/issue/{issue_key}/properties", func=lambda v: v["keys"], params={}
+            ):
+                issue_property_key = issue_property_key_item.get("key")
+                for issue_property in self.lists(
```

Comment:
isn't this extracting the `keys` property and not `value` property? this is a bit confusing. I don't think this is emitting the correct schema.

Reply:
In case of Issue properties stream the first step would be taking property [keys](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-properties/#api-rest-api-3-issue-issueidorkey-properties-get) and then directly properties.